src/main/java/core/basesyntax/service/impl/ParserImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Parser;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ParserImpl implements Parser {
+    @Override
+    public List<FruitTransaction> parse(List<String[]> lines) {
+        List<FruitTransaction> commands = new ArrayList<>();
+        for (String[] line : lines) {
+            if (line.length != 3) {
+                throw new RuntimeException("Can't parse information from " + Arrays.toString(line));
```

Comment:
It's good that you're checking the length of the line array, but consider creating a custom exception that more accurately reflects the domain of the application, such as `ParseException`. This would be more informative than a generic `RuntimeException`.